docs/excel/custom-functions-best-practices.md

```diff
@@ -125,6 +125,63 @@ Keep in mind the following best practices when creating custom functions in your
     }
     ```
 
+## Declaring optional parameters 
+In Excel for Windows (version 1812 or later), you can declare optional parameters for your custom functions. When a user invokes a function in Excel, optional parameters appear in brackets. For example, a function `FOO` with one required parameter called `parameter1` and one optional parameter called `parameter2` would appear as `=FOO(parameter1, [parameter2])` in Excel.
+
+To make a parameter optional, add `"optional": "true"` to the parameter in the JSON metadata file that defines the function. The following example shows what this might look like for the function `=ADD(first, second, [third])`. Notice that the optional `[third]` parameter follows the two required parameters. Required parameters will appear first in Excel’s Formula UI.
+
+```json
+{
+    "id": "add",
+    "name": "ADD",
+    "description": "Add two numbers",
+    "helpUrl": "http://www.contoso.com",
+    "result": {
+        "type": "number",
+        "dimensionality": "scalar"
+        },
+    "parameters": [
+        {
+            "name": "first",
+            "description": "first number to add",
+            "type": "number",
+            "dimensionality": "scalar"
+        },
+        {
+            "name": "second",
+            "description": "second number to add",
+            "type": "number",
+            "dimensionality": "scalar",
+        },
+        {
+            "name": "third",
+            "description": "third optional number to add",
+            "type": "number",
+            "dimensionality": "scalar",
+            "optional": true
+        }
+    ],
+    "options": {
+        "sync": false
+    }
+}
+```
+
+When you define a function that contains one or more optional parameters, you should specify what happens when the optional parameters are undefined. You may do so by using an `if` statement or by using the JavaScript OR operator. In the following example, `zipCode` and `dayOfWeek` are both optional parameters for the `getWeatherReport` function. If the `zipCode` parameter is undefined, the default value is set to 98052. If the `dayOfWeek` parameter is undefined, it is set to Wednesday.
+
+```js
+function getWeatherReport(required, zipCode, dayOfWeek)
+{
+  if (zipCode === undefined) {
+      zipCode === 98052
+  }
+  if (dayOfWeek === undefined) {
+    dayOfWeek === "Wednesday"
+}
+    ... //After establishing parameters, this function would have additional code to get a weather report
+}
```

Comment:
Multiple issues with this code snippet:
- the first input parameter (`required`) is not used and doesn't make any sense in this context 
- incorrect use of triple-equals-signs for assigning value to a variable
- missing semi-colons
- inconsistent use of indentation: 4 spaces on some lines, 2 spaces on other lines
- closing brace on the second `if` statement is missing indentation
- awkward positioning and wording of code comment

Suggest replacing the code snippet with the following  (and then verifying what the snippet looks like on the staging site after you commit/push and the OPS build completes):

```
function getWeatherReport(zipCode, dayOfWeek)
{
  if (zipCode === undefined) {
    zipCode = "98052";
  }

  if (dayOfWeek === undefined) {
    dayOfWeek = "Wednesday";
  }

  // Get weather report for specified zipCode and dayOfWeek
  // ... 
}
```

Reply:
@mscharlock looks like you applied some of the suggested fixes to this [code sample](https://review.docs.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-best-practices?branch=michelle-optionalparams#declaring-optional-parameters), but it still needs a couple more tweaks. 

![image](https://user-images.githubusercontent.com/3019529/49305707-d8811a00-f484-11e8-83f6-6faefb2a04e1.png)

- add semicolon to the end of these lines:
    - `zipCode = "98052";`
    - `dayOfWeek = "Wednesday";`
- fix indentation (outdent by 2 spaces) for the last three lines before the final brace:
    - `}`
    - `// Get weather report for specified zipCode and dayOfWeek`
    - `// ...`

Reply:
Yes, sorry about this sample - I think I was rushing through things too quickly yesterday and had to get out the door to get home to meet someone so I didn't get all the way through (hence, this is still unresolved). Will fix this am.